### PR TITLE
DryRunMemory

### LINKED ISSUE #1538: Javascript Dry Run memory
Not sure if it is an issue or intended, i was just confused at first :smile: . 

When i dry run a Javascript Agent in agents/show and agents/edit with 
```
Agent.check = function() {
  this.createEvent({"out": this.memory("test")});
};
```
and the memory of the Agent before the dry run is set to
`{ "test": "value" }` i get different results. 

The edit version creates an empty event while the show version creates 
`[ { "memory": "value" } ]`.
The edit version doesn't seem to remember previous set memory keys.

### PR DESCRIPTION
**Pull request by Enfop**
With the PR i try to fix #1538. The changes fix the issue but i'm not sure if other agents are affected in a negative way.

### CODE REVIEW

File: spec/controllers/agents/dry_runs_controller_spec.rb [hunk 1]
```diff
@@ -100,5 +100,22 @@ def valid_attributes(options = {})
...
     end
+
+    it "uses the memory of an existing Agent" do
+      valid_params = {
+        :name => "somename",
+        :options => {
+          :code => "Agent.check = function() { this.createEvent({ 'message': this.memory('fu') }); };",
+        }
+      }
+      agent = Agents::JavaScriptAgent.new(valid_params)
```
Comment (dsander) [inline spec/controllers/agents/dry_runs_controller_spec.rb]: Something minor, you could do `agent = Agent.build_for_type(Agents::JavaScriptAgent, users(:bob), memory: {fu: 'bar'})` to save a few lines.

**Comment by dsander** [comment]
Thanks @Enfop, looks good to me. I do not think there are negative effects for other agents, as I see it it only makes the dry-run match the actual run more closely.

**Comment by cantino** [comment]
Makes sense to me too.  I wonder if we should eventually allow you to edit the memory in the dry run UI.

**Comment by dsander** [comment]
@cantino For advanced `JavaScriptAgent` use cases that would be neat, don't know how many other agents could benefit from it.

**Comment by cantino** [comment]
I'm going to go ahead and merge this.  Thanks @Enfop!

**Pull request merged**